CedarJava/src/test/java/com/cedarpolicy/EntityValidationTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022-2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cedarpolicy;
+
+import static com.cedarpolicy.TestUtil.loadSchemaResource;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeAll;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+
+import com.cedarpolicy.model.EntityValidationRequest;
+import com.cedarpolicy.model.EntityValidationResponse;
+
+import com.cedarpolicy.model.exception.AuthException;
+import com.cedarpolicy.model.schema.Schema;
+import com.cedarpolicy.model.slice.Entity;
+import com.cedarpolicy.pbt.EntityGen;
+import com.cedarpolicy.value.EntityTypeName;
+import com.cedarpolicy.value.PrimBool;
+
+
+/**
+ * Tests for entity validator
+ */
+public class EntityValidationTests {
+    private static EntityGen entityGen;
+    private static AuthorizationEngine engine;
+
+    /**
+     * Test that a valid entity is accepted.
+     */
+    @Test
+    public void testValidEntity() throws AuthException {
+        Entity entity = EntityValidationTests.entityGen.arbitraryEntity();
+
+        EntityValidationRequest r = new EntityValidationRequest(
+                ROLE_SCHEMA, List.of(entity));
+
+        EntityValidationResponse result = engine.validateEntities(r);
+
+        assertEquals(result.getMessage(), "Entities parsed correctly");
+    }
+
+    /**
+     * Test that an entity with an attribute not specified in the schema throws an exception.
+     */
+    @Test
+    public void testEntityWithUnknownAttribute() throws AuthException {
+        Entity entity = EntityValidationTests.entityGen.arbitraryEntity();
+        entity.attrs.put("test", new PrimBool(true));
+
+        EntityValidationRequest request = new EntityValidationRequest(
+                ROLE_SCHEMA, List.of(entity));
+
+
+        AuthException exception = assertThrows(AuthException.class, () -> engine.validateEntities(request));
```

Comment:
I was wondering if this is the exception that should be thrown for this test.

Reply:
`InternalException` seems like the best fit.

Reply:
`BadRequestException` is fine, or you could make a custom exception, along the lines of `InvalidEUIDException`.